core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/GatewayConfigProvider.java

```diff
@@ -0,0 +1,29 @@
+package com.sequenceiq.cloudbreak.service.cluster;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.base.InstanceMetadataType;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceMetaData;
+import com.sequenceiq.cloudbreak.orchestrator.model.GatewayConfig;
+import com.sequenceiq.cloudbreak.service.GatewayConfigService;
+
+@Component
+public class GatewayConfigProvider {
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    public GatewayConfig getGatewayConfig(Stack stack) {
```

Comment:
why don't you use `GatewayConfigService#getPimaryGatewayConfig`?

Reply:
I have extracted this from InstanceMetadataUpdater, but if both of them are effectively the same, I can use that.